lib/segment/src/index/query_optimization/condition_converter.rs

```diff
@@ -121,6 +121,17 @@ pub fn condition_converter<'a>(
                 })
             })
         }
+        Condition::Resharding(cond) => {
+            let external_ids: HashMap<_, _> = id_tracker
+                .iter_internal()
+                .filter_map(|internal_id| Some((internal_id, id_tracker.external_id(internal_id)?)))
+                .collect();
+
+            Box::new(move |point_id| match external_ids.get(&point_id) {
+                Some(&point_id) => cond.check(point_id),
+                None => false,
+            })
+        }
```

Comment:
How about pre-filtering here? Checking the hashring condition when building the external ID list.

Reply:
Hmmmmmm. Maybe. Let me think about this for a moment.

Reply:
Done.